src/net/telnet.cc

```diff
@@ -293,7 +293,8 @@ static inline void on_telnet_subnegotiation(unsigned char cmd, const char *buf,
       char *str = new_string(size, "telnet suboption");
       str[size] = '\0';
       for (int i = 0; i < size; i++) {
-        str[i] = (buf[i] ? buf[1] : 'I');
+        // str[i] = (buf[i] ? buf[1] : 'I'); <- propagates 2nd element of buf
```

Comment:
You should just remove the wrong code.

I'm not sure if this was my typo.... it doesn't seem the type of the mistake I would make.... :-p